limited_area/region_spec.py

```diff
@@ -165,3 +184,62 @@ def circle(self, center_lat, center_lon, radius):
 
         return np.array(ll)
 
+    def ellipse(self, center_lat, center_lon, semi_major, semi_minor, orientation):
+        """ Return a list of points that form an ellipse around [center_lat, center_lon] 
+
+        center_lat - The center latitude of the ellipse in degrees
+        center_lon - The center longitude of the ellipse in degrees
+        semi_major - The length of the semi_major axies in meters
+        semi_minor - The legnth of the semi_minor axies in meters
+        orientaiton - The orientation of the desired ellipse, rotated clockwise from north.
+
+        Given the center_lat and center_lon of an ellipse, create a region that forms an ellipse
+        with the semi major axies length being == `semi_major` and the semi minor axies being ==
+        `semi_minor` and rotate the ellipse clockwise by `orientation` from due North.
+
+        """
+
+        P = []
+
+        # Convert ellipse center from (lat,lon) to Cartesian
+        C = latlon_to_xyz(center_lat, center_lon, 1.0)
+
+        # Convert semi-major and semi-minor axis lengths from meters to radians
+        semi_major = semi_major / EARTH_RADIUS
+        semi_minor = semi_minor / EARTH_RADIUS
+
+        a = semi_major
+        b = semi_minor
+
+        # Convert orientation angle to radians
+        orientation = orientation * np.pi / 180.0
+
+        # Find a point not equal to C or -C
+        K = np.zeros(3)
+        K[0] = 0
+        K[1] = 0
+        K[2] = 1
+
+        if abs(np.dot(K,C)) >= 0.9:
+            K[0] = 1
+            K[1] = 0
+            K[2] = 0
+
+        # S is then a vector orthogonal to C
+        S = np.cross(C, K)
+        S = S / np.linalg.norm(S)
+
+        for r in np.linspace(0.0, 2.0*np.pi, 100):
+            radius = np.sqrt((semi_major * np.cos(r))**2 + (semi_minor * np.sin(r))**2)
+            P0 = rotate_about_vector(C, S, radius)
+            d = rotate_about_vector(P0, C, r)
```

Comment:
It looks like I also forgot to delete `d` -- it was presumably just used for debugging purposes.